gradle/javadoc.gradle

```diff
@@ -90,5 +91,18 @@ allprojects {
                     |
                     | </script>'''.stripMargin()
         }
+
+        afterEvaluate {
+            destinationDir = new File(rootDir, "build/docs/${project.archivesBaseName}")
+            def version = project.version as String
+            if (project.name != 'bson') {
+                options.linksOffline(*buildDocsUrl(version, 'bson'))
+                if (project.name != 'driver-core') options.linksOffline(*buildDocsUrl(version, 'mongodb-driver-core'))
+                if (project.name == 'driver-legacy') {
+                    dependsOn javaMainProjects.find({ it.name == 'driver-sync' }).javadoc
```

Comment:
Why do we only need dependsOn for driver-sync from driver-legacy?

Reply:
This is to do with the ordering of the build and ensures the javadoc is built and can be read then linked to.